### PR TITLE
Update Scanner file

### PR DESCRIPTION
**Pull request by GMNBNBNB**

### CODE REVIEW

File: tokenizer/scanner.py [hunk 1]
```diff
@@ -1,3 +1,78 @@
...
+                                i += len(keyword)
+                                matched = True
+                                break
+                        if matched:
+                            break
+
+                # Operator
+                if not matched:
+                    for operator in token_specification[1][1]:
+                        if input_string[i:i+len(operator)] == operator:
```
Comment (Yixli9811) [inline tokenizer/scanner.py:41]: There might be a problem here: if operator is "==", "=" will be matches first, and then break out of the loop

File: tokenizer/scanner.py [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+                    i += 1
+                    continue
+
+                matched = False
+
+                # Keyword
+                for token_type, values in token_specification:
+                    if token_type == 'KEYWORD':
+                        for keyword in values:
+                            if input_string[i:i+len(keyword)] == keyword and (i+len(keyword) == len(input_string) or not input_string[i+len(keyword)].isalnum()):
```
Comment (Yixli9811) [inline tokenizer/scanner.py:30]: What's the logic of the second and third condition?

File: tokenizer/scanner.py [hunk 3]
```diff
@@ -1,3 +1,78 @@
 
-def scan(code):
-    print("scanner")
+def scan(filename):
+    print("scanner start")
+    token_specification = [
+        ('KEYWORD', ['if', 'then', 'else', 'while', 'do', 'end', 'procedure', 'var', 'begin', 'print']),
+        ('OPERATOR', ['=','+', '-', '*', '/', '==', '!=', '<=', '>=', '<', '>', 'and', 'or', 'not']),
+        ('SEPARATOR', [';']),
+        ('LPAR', ['(']),
```
Comment (Yixli9811) [inline tokenizer/scanner.py:8]: We still need to do LPAR and RPAR right?

File: tokenizer/scanner.py (hunk 2 above)
Comment (GMNBNBNB) [reply tokenizer/scanner.py:30]: Check if it is the last word of the sentence. If it is, no additional check is needed. If it is not, check if any other word follows it. For example, 'while' is a keyword, but 'while1' is not.

File: tokenizer/scanner.py (hunk 1 above)
Comment (GMNBNBNB) [reply tokenizer/scanner.py:41]: This is indeed a problem, I will fix it later.
(cant check ==)

File: tokenizer/scanner.py (hunk 3 above)
Comment (GMNBNBNB) [reply tokenizer/scanner.py:8]: It include in Symbol check

Comment (Yixli9811) [reply tokenizer/scanner.py:8]: Oh I see it now

File: tokenizer/scanner.py (hunk 2 above)
Comment (Yixli9811) [reply tokenizer/scanner.py:30]: Hmm so when there is a special character in the string e.g. "while@", this will accept it right?

**Review by Yixli9811 — APPROVED** [review]

**Comment by GMNBNBNB** [comment]
# [TO DO]: Condition ==

**Pull request merged**